src/extension/wps/src/main/java/au/org/emii/aggregator/template/TemplateDataset.java

```diff
@@ -39,7 +39,7 @@ public TemplateDataset(NetcdfDatasetIF dataset, Set<String> requestedVariables,
         String timeDimension = dataset.getTimeAxis().getDimensionName();
 
         for (NetcdfVariable variable: dataset.getVariables()) {
-            if (requestedVariables != null && requestedVariables.contains(variable.getShortName())) {
+            if (requestedVariables != null && !requestedVariables.contains(variable.getShortName())) {
```

Comment:
I'm only including dimensions in the output dataset which are used in the selected variables and this may not be all the dimensions in the original dataset.